detectable/src/main/java/com/synopsys/integration/detectable/detectables/gradle/inspection/parse/GradleReportLineParser.java

```diff
@@ -101,26 +107,22 @@ private List<String> parseGav(String line, Map<String, String> metadata) {
         if(gavPieces.size() == 3) {
             String dependencyGroupName = gavPieces.get(0) + ":" + gavPieces.get(1);
             if ((cleanedOutput.contains(STRICTLY) || cleanedOutput.contains(REJECT) || cleanedOutput.contains(REQUIRE) || cleanedOutput.contains(PREFER))) {
-                gradleRichVersions.putIfAbsent(dependencyGroupName, gavPieces.get(2));
-                richVersionGroup.computeIfAbsent(projectName, value -> new HashSet<>()).add(dependencyGroupName);
-            }
-
-            if(gradleRichVersions.containsKey(dependencyGroupName)) {
-                updateRichVersion(dependencyGroupName, projectParent, projectName, rootProjectName, gavPieces);
+                gradleRichVersions.computeIfAbsent(projectName, value -> new HashMap<>()).putIfAbsent(dependencyGroupName, gavPieces.get(2));
```

Comment:
Why not Map of Set (or List if order is important)?

Reply:
Earlier, I had map of set there but I think Map of Map can help to store the version which was declared. We would need it while execution.